tests/test_stream.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+import hashlib
+import os
+import shutil
+import tempfile
+from typing import Optional
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+
+from streaming import Stream
+from streaming.base.distributed import barrier
+
+
+@pytest.mark.world_size(2)
```

Comment:
I am not 100% sure that by setting `@pytest.mark.world_size(2)` would create 2 sub-processes. Can you check once by printing the `RANK`, `WORLD_SIZE`, and `LOCAL_WORLD_SIZE` ?

Reply:
Yep, shows the correct world size.